packages/web/src/js/Collapse.ts

```diff
@@ -0,0 +1,167 @@
+import BaseComponent from './BaseComponent';
+import { enableToggleAutoloader } from './utils/ComponentFunctions';
+import selectorEngine from './dom/SelectorEngine';
+import EventHandler from './dom/EventHandler';
+
+const NAME = 'collapse';
+const NAME_ARIA_EXPANDED = 'aria-expanded';
+const NAME_ARIA_CONTROLS = 'aria-controls';
+const NAME_DATA_TARGET = 'data-target';
+const CLASSNAME_EXPANDED = 'is-expanded';
+const CLASSNAME_COLLAPSED = 'is-collapsed';
+
+enum breakpointKeys {
+  default = 'default',
+  mobile = 'mobile',
+  tablet = 'tablet',
+  desktop = 'desktop',
+}
+type breakpointType = keyof typeof breakpointKeys;
+
+interface CollapseMeta {
+  id: string | 'unknown';
+  hideOnCollapse: boolean;
+  breakpoint: breakpointType;
+}
+interface CollapseState {
+  collapsed: boolean;
+  width: number;
+  triggered: boolean;
+}
+
+class Collapse extends BaseComponent {
+  target: HTMLElement | null | undefined;
+  parent: HTMLElement | null | undefined;
+  meta: CollapseMeta;
+  state: CollapseState;
+
+  constructor(element: HTMLElement) {
+    super(element);
+    this.target = this.element.dataset.target ? selectorEngine.findOne(`#${this.element.dataset.target}`) : null;
+    this.parent = this.element && this.element.parentElement;
+    this.meta = {
+      id: this.element.dataset.target,
+      hideOnCollapse: !!(this.element.dataset.more || this.element.dataset.more === ''),
+      breakpoint: (this.target?.dataset?.breakpoint as breakpointType) || breakpointKeys.default,
+    };
+    this.state = {
+      collapsed: !!(
```

Comment:
I would rather use `this.isCollapsed` and `this.isTriggered` instead of trying to simulate the usage of state like in React.